vivarium/experiments/test_profiler.py

```diff
@@ -0,0 +1,59 @@
+import time
+
+from vivarium.core.engine import Engine
+from vivarium.core.process import Process
+from vivarium.core.types import Schema, State, Update
+
+
+class ProcessA(Process):
+
+    defaults = {
+        'sleep': 2,
+    }
+
+    def ports_schema(self) -> Schema:
+        return {}
+
+    def next_update(self, timestep: float, states: State) -> Update:
+        time.sleep(self.parameters['sleep'])
+        return {}
+
+
+class ProcessB(Process):
+
+    defaults = {
+        'sleep': 1,
+        '_parallel': True,
+    }
+
+    def ports_schema(self) -> Schema:
+        return {}
+
+    def next_update(self, timestep: float, states: State) -> Update:
+        time.sleep(self.parameters['sleep'])
+        return {}
+
+
+def test_profiler() -> None:
+    engine = Engine(
+        processes={
+            'processA': ProcessA(),
+            'processB': ProcessB(),
+        },
+        topology={
+            'processA': {},
+            'processB': {},
+        },
+        profile=True,
+    )
+    engine.update(3)
+    engine.end()
+    assert engine.stats is not None
+    stats = engine.stats.strip_dirs()
+    process_a_runtime = stats.stats[  # type: ignore
+        ('test_profiler.py', 16, 'next_update')][3]
+    process_b_runtime = stats.stats[  # type: ignore
+        ('test_profiler.py', 31, 'next_update')][3]
```

Comment:
Out of curiosity, what are those `16`, `31` for?

Reply:
Those are the line numbers for where the two `next_update` functions are defined. That's how `cProfile` resolves cases like this where there are 2 functions of the same name in a file